src/renderers/shared/fiber/isomorphic/ReactPortal.js

```diff
@@ -27,6 +27,7 @@ exports.createPortal = function(
   implementation: any,
   key: ?string = null,
 ): ReactPortal {
+  containerInfo._reactPortalContainer = {};
```

Comment:
Good points;
 - Absolutely we should check for the presence of '_reactPortalContainer' before adding another one. Will fix.
 - I can look for a place add the '_reactPortalContainer' that is more similar to where the '_reactRootContainer' is added.
 - Will wrap the '_reactPortalContainer' addition in a `__DEV__` check.
 - I could rename '_reactPortalContainer' to 'unstable_reactPortalContainer' to additionally show that it's not to be relied on.
 - Agreed in general that it feels odd; if there is another way to determine whether the parent is a portal I'd rather use existing code, but didn't see anything else that was exposed there.

> I also wonder if this means `createPortal(<div />, null)` is going to throw (because `containerInfo` is `null` but is being accessed). Which I think ideally it should, but I’m not sure if this is the case now.

That sounds about right to me, and I also would expect an error if I tried to pass 'null' as the container to `createPortal`. Hopefully this is a corner case, and if product code relied on it they could add a `null` check.

We only have [one internal use case](https://fburl.com/internal_use_of_portal) and  in that framework they currently create a new `div` element for the `portalContainer` and then they do set it to `null` when unmounting the component.

So if somehow `createPortal` was called after the portal component unmounted that would be a problem. I don't think that is possible with the current code but maybe it's worth adding a warning?